src/abstractdataframe/abstractdataframe.jl

```diff
@@ -827,29 +843,25 @@ function _vcat(dfs::AbstractVector{<:AbstractDataFrame})
                 estrings[i] = "column(s) $cols are missing from argument(s) $args"
             end
             throw(ArgumentError(join(estrings, ", ", ", and ")))
-        else
-            estrings = Vector{String}(length(uniqueheaders))
-            for (i, u) in enumerate(uniqueheaders)
-                indices = find(a -> a == u, allheaders)
-                estrings[i] = "column order of argument(s) $(join(indices, ", ", " and "))"
-            end
-            throw(ArgumentError(join(estrings, " != ")))
         end
-    else
-        header = uniqueheaders[1]
-        cols = Vector{Any}(length(header))
-        for i in 1:length(cols)
-            data = [df[i] for df in dfs]
-            lens = map(length, data)
-            cols[i] = promote_col_type(data...)(sum(lens))
-            offset = 1
-            for j in 1:length(data)
-                copy!(cols[i], offset, data[j])
-                offset += lens[j]
-            end
+        # this should not happen
+        throw(ArgumentError("Unexpected error - please report steps allowing to reproduce it"))
```

Comment:
Better do `@assert !isempty(coldiff)`, as `ArgumentError` is supposed to reflect an error made by the caller. Or just remove the check, as it could only happen if there was a bug in Base, and we can't check all possible bugs... ;-)

Reply:
Finally I have made a bit different implementation based on `coldiff` without using `Set`s which should be safe.